src/main/java/com/microfocus/octane/plugins/rest/RestConnector.java

```diff
@@ -313,33 +313,39 @@ private Response retrieveHtmlResponse(HttpURLConnection con) throws IOException
 
         Response ret = new Response();
 
-        InputStream inputStream;
+        InputStream inputStream = null;
         //select the source of the input bytes, first try "regular" input
         try {
-            inputStream = con.getInputStream();
-        } catch (Exception e) {
+            inputStream = (con.getResponseCode() < HttpURLConnection.HTTP_BAD_REQUEST) ? con.getInputStream() : con.getErrorStream();
+            //this actually takes the data from the previously decided stream (error or input) and stores it in a byte[] inside the response
+            if (inputStream != null) {
+                ByteArrayOutputStream container = new ByteArrayOutputStream();
+
+                byte[] buf = new byte[1024];
+                int read;
+                while ((read = inputStream.read(buf, 0, 1024)) > 0) {
+                    container.write(buf, 0, read);
+                }
+
+                ret.setResponseData(container.toString(Charsets.UTF_8));
+            }
+
+        } catch (IOException e) {
             /*if the connection to the server somehow failed, for example 404 or 500, con.getInputStream() will throw an exception, which we'll keep.
             we'll also store the body of the exception page, in the response data. */
-            inputStream = con.getErrorStream();
             ret.setFailure(e);
             ret.setResponseData(e.getMessage());//set default error message
-        }
-
-        //this actually takes the data from the previously decided stream (error or input) and stores it in a byte[] inside the response
-        if (inputStream != null) {
-            ByteArrayOutputStream container = new ByteArrayOutputStream();
-
-            byte[] buf = new byte[1024];
-            int read;
-            while ((read = inputStream.read(buf, 0, 1024)) > 0) {
-                container.write(buf, 0, read);
+        } finally {
+            if (inputStream != null) {
```

Comment:
in try block you already have a check for inputStream != null - can't this be avoided here?

Reply:
you can use IOUtils.closequietly to not use !=null here